warehouse/slave_test.go

```diff
@@ -1,254 +1,232 @@
+		DestinationRevisionID:        uuid.New().String(),
+		StagingDestinationRevisionID: uuid.New().String(),
+		DestinationConfig:            destConf,
+		StagingDestinationConfig:     map[string]interface{}{},
+		UniqueLoadGenID:              uuid.New().String(),
+		RudderStoragePrefix:          misc.GetRudderObjectStoragePrefix(),
+		LoadFileType:                 "csv",
+	}
+
+	payloadJson, err := json.Marshal(p)
 	require.NoError(t, err)
 
-	ctxCancel, cancel := context.WithCancel(context.Background())
-	cancel()
-
-	provider := "MINIO"
-	workspaceID := "test-workspace-id"
-	destinationID := "test-destination-id"
-	destinationName := "test-destination-name"
-	sourceID := "test-source-id"
-	sourceName := "test-source-name"
-	destType := "POSTGRES"
-	worker := 7
-	prefix := warehouseutils.DatalakeTimeWindowFormat
-	namespace := "test-namespace"
-	loadObjectFolder := "test-load-object-folder"
-
-	testCases := []struct {
-		name              string
-		destType          string
-		ctx               context.Context
-		conf              map[string]any
-		additionalWriters int
-		wantError         error
-	}{
-		{
-			name:              "Parquet file",
-			additionalWriters: 9,
-			destType:          warehouseutils.S3Datalake,
-		},
-		{
-			name: "Few files",
-		},
-		{
-			name:              "many files",
-			additionalWriters: 49,
-		},
-		{
-			name:      "Context cancelled",
-			ctx:       ctxCancel,
-			wantError: errors.New("uploading load file to object storage: context canceled"),
-		},
-		{
-			name: "Unknown provider",
-			conf: map[string]any{
-				"bucketProvider": "UNKNOWN",
-			},
-			wantError: errors.New("creating uploader: service provider not supported: UNKNOWN"),
-		},
-		{
-			name: "Invalid endpoint",
-			conf: map[string]any{
-				"endPoint": "http://localhost:1234",
-			},
-			wantError:         errors.New("uploading load file to object storage: uploading load file: Endpoint url cannot have fully qualified paths."),
-			additionalWriters: 9,
-		},
+	claim := pgnotifier.Claim{
+		ID:        1,
+		BatchID:   uuid.New().String(),
+		Payload:   payloadJson,
+		Status:    "waiting",
+		Workspace: "test_workspace",
+		JobType:   "upload",
 	}
 
-	for _, tc := range testCases {
-		tc := tc
-
-		t.Run(tc.name, func(t *testing.T) {
-			minioResource, err := destination.SetupMINIO(pool, t)
-			require.NoError(t, err)
-
-			conf := map[string]any{
-				"bucketName":       minioResource.BucketName,
-				"accessKeyID":      minioResource.AccessKey,
-				"accessKey":        minioResource.AccessKey,
-				"secretAccessKey":  minioResource.SecretKey,
-				"endPoint":         minioResource.Endpoint,
-				"forcePathStyle":   true,
-				"s3ForcePathStyle": true,
-				"disableSSL":       true,
-				"region":           minioResource.SiteRegion,
-				"enableSSE":        false,
-				"bucketProvider":   provider,
-			}
+	go func() {
+		for i := 0; i < workerJobs; i++ {
+			subscriberCh <- claim
+		}
+	}()
```

Comment:
Should we wait for this go routine  to return before the test function concludes?

Reply:
We are waiting below `response := <-publishCh`